launcher/ui/widgets/JavaWizardWidget.h

```diff
@@ -0,0 +1,103 @@
+#pragma once
+#include <QWidget>
+
+#include <BaseVersion.h>
+#include <QObjectPtr.h>
+#include <java/JavaChecker.h>
+#include <qcheckbox.h>
+#include <QIcon>
+
+class QLineEdit;
+class VersionSelectWidget;
+class QSpinBox;
+class QPushButton;
+class QVBoxLayout;
+class QHBoxLayout;
+class QGroupBox;
+class QGridLayout;
+class QLabel;
+class QToolButton;
+class QSpacerItem;
+
+class JavaWizardWidget : public QWidget {
+    Q_OBJECT
+
+   public:
+    explicit JavaWizardWidget(QWidget* parent);
+    virtual ~JavaWizardWidget();
+
+    enum class JavaStatus { NotSet, Pending, Good, DoesNotExist, DoesNotStart, ReturnedInvalidData } javaStatus = JavaStatus::NotSet;
+
+    enum class ValidationStatus { Bad, JavaBad, AllOK };
+
+    void refresh();
+    void initialize();
+    ValidationStatus validate();
+    void retranslate();
+
+    bool permGenEnabled() const;
+    int permGenSize() const;
+    int minHeapSize() const;
+    int maxHeapSize() const;
+    QString javaPath() const;
+    bool autoDetectJava() const;
+    bool autoDownloadJava() const;
+
+    void updateThresholds();
+
+   protected slots:
+    void onSpinBoxValueChanged(int);
+    void memoryValueChanged();
+    void javaPathEdited(const QString& path);
+    void javaVersionSelected(BaseVersion::Ptr version);
+    void on_javaBrowseBtn_clicked();
+    void on_javaStatusBtn_clicked();
+    void javaDownloadBtn_clicked();
+    void checkFinished(const JavaChecker::Result& result);
+
+   protected: /* methods */
+    void checkJavaPathOnEdit(const QString& path);
+    void checkJavaPath(const QString& path);
+    void setJavaStatus(JavaStatus status);
+    void setupUi();
+
+   private: /* data */
+    VersionSelectWidget* m_versionWidget = nullptr;
```

Comment:
Why is the ui here not defined in a UI file?
Also, the destructor only deletes a spacer, is this intended?

Reply:
This is just a renamed file, fixing all of it's problems is not something I want to deal with in this PR... I did not merge it with JavaSettingsWidget as it behaves so differently

Reply:
Fine by me(still this would be nice)